### PR TITLE
Fix CircleCI error caused by GraphQL change to GraphAPI.FlaggingKind

### PR DESCRIPTION
**Pull request by amy-at-kickstarter**
# 📲 What

* Add graphql-schema.json to dependency list for Apollo update script
* Change type of CreateFlaggingInput from GraphAPI.FlaggingKind to GraphAPI.NonDeprecatedFlaggingKind

# 🤔 Why

Ingerid encountered this issue in a CircleCI build. Looks like we were running the Apollo update script on Circle, and it was making changes to `GraphAPI.swift` which broke her build. This fixes both the change to the schema, as well as makes sure that future updates to the schema will correctly cause a re-run of the Apollo build script.

### CODE REVIEW

File: KsApi/GraphAPI.swift [hunk 1]
```diff
@@ -810,7 +805,7 @@ public enum GraphAPI {
       }
     }
 
-    public static var allCases: [FlaggingKind] {
+    public static var allCases: [NonDeprecatedFlaggingKind] {
```
Comment (amy-at-kickstarter) [inline KsApi/GraphAPI.swift:808]: This rename is the source of the problem.

**Review by ifosli — APPROVED** [review]

File: Library/ReportProjectInfoListItem.swift [hunk 2]
```diff
@@ -84,7 +84,7 @@ let reportSpamSubListItems = [
   ),
   ReportProjectInfoListItem(
     type: .child,
-    flaggingKind: .abuse,
+    flaggingKind: .guidelinesAbuse,
```
Comment (amy-at-kickstarter) [inline Library/ReportProjectInfoListItem.swift:87]: @scottkicks This is the one material change here - `NonDeprecatedFlaggingKind` does not include `.abuse` but it does include `.guidelinesAbuse`. What are these enums ultimately used for? Does it matter?

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/ReportProjectInfoListItem.swift (hunk 2 above)
Comment (amy-at-kickstarter) [reply Library/ReportProjectInfoListItem.swift:87]: I just tested it on staging and was able to submit a report - I tried IP and Abuse. Both look sane:
Screenshot 2024-03-19 at 4 48 28 PM

Comment (scottkicks) [reply Library/ReportProjectInfoListItem.swift:87]: They're used to determine the category reason why users are reporting. I think this should be fine

**Review by scottkicks — APPROVED** [review]

**Pull request merged**